### PR TITLE
bugfix/1329 - Free up CIDs after use

### LINKED ISSUE #1329: CID values not properly freed up from _cidNumbersInUse
Issue title has been updated to reflect root cause of problem rather than just reported symptoms.
Details given comment [further down](#issuecomment-467923344).

Original bug report follows:

---

Browser: Chrome

Airport: KBOS

Approximate play time before issue:
Playing on-and-off over the course of 24h.
Browser was left open in background when not playing.

#### Issue Description
I've had this happen three times now:  
After playing for a long duration, game halts completely and does not respond to any inputs. Checking the console reveals large (and growing) number of "Uncaught RangeError: Maximum call stack size exceeded" which continue to accumulate unless script execution is halted (F8).

For a rough ballpark, this occurs at a score of >9000 but <10,000.  
Most recently happened at 9610 score with console output flooded with:

openscopecallstack

Example call stack:

> value (StripViewController.js:290)
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:332)
... (repeats many, many times) ...
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:332)
value (StripViewController.js:177)
value (StripViewController.js:161)
value (AircraftController.js:426)
value (AppController.js:252)
value (App.js:301)
value (App.js:324)
requestAnimationFrame (async)
value (App.js:321)
requestAnimationFrame (async)
value (App.js:321)
requestAnimationFrame (async)
value (App.js:321)
requestAnimationFrame (async)
value (App.js:321)
requestAnimationFrame (async)
value (App.js:321)
... (repeats many times)

#### Steps to reproduce
Play for a long time

### PR DESCRIPTION
**Pull request by cake-pie**
Resolves #1329

The purpose of this pull request is to properly free up CIDs when they are no longer being used.

### CODE REVIEW

**Comment by cake-pie** [comment]
> store cid as an actual number and then add a new getter that the view can use that handles the `leftPad` work.

Hmm, otoh a strip view can only be removed once, but could be redrawn multiple times.
How about storing cid as both string and number, to avoid repeated calls to `leftPad`?

**Comment by n8rzz** [comment]
@cake-pie good call!

File: src/assets/scripts/client/aircraft/StripView/StripViewModel.js [hunk 1]
```diff
@@ -153,17 +153,28 @@ export default class StripViewModel extends BaseModel {
...
-         * in the case a value is less than 100
-         *
          * Currently only numeric, but could be made alpha-numberic
          *
+         * Padded with leading zeros in `#cidString` for display purposes
+         *
          * @property cid
          * @type {number}
          * @default cidValue
          * @private
```
Comment (felixscheffer) [inline src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: Can you remove the private please, as it's not private

File: src/assets/scripts/client/aircraft/StripView/StripViewModel.js [hunk 2]
```diff
@@ -153,17 +153,28 @@ export default class StripViewModel extends BaseModel {
...
+        /**
+         * A string version of `#cid` for display purposes
+         * Has been padded with leading zeros so that it is always
+         * displayed as three digits
+         *
+         * @property cidString
+         * @type {string}
+         * @default cidValue
+         * @private
+         */
```
Comment (felixscheffer) [inline src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: Can you remove the private please, as it's not private

**Pull request merged**